DatadogRUM/Sources/Instrumentation/AppHangs/AppHangsWatchdogThread.swift

```diff
@@ -0,0 +1,131 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+internal struct AppHang {
+    /// The date of hang end.
+    let date: Date
```

Comment:
```suggestion
    let endTime: Date
```

I kind of prefer having startTime and duration in general, endTime and duration seems odd from API point of view.

Reply:
If not blocking, I'd keep it as it is. Hangs are reported as RUM Errors, so they have no duration on view's timeline and we want to have them reported at the time of hang end.

Hang error will follow the Long Task that visualizes the duration:
<img width="220" alt="Screenshot 2024-02-23 at 12 33 59" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/04388959-eb25-4203-a942-2ddd77bb5fb0">